cmd/snap-repair/runner.go

```diff
@@ -655,14 +654,65 @@ func verifySignatures(a asserts.Assertion, workBS asserts.Backstore, trusted ass
 	return nil
 }
 
-func (run *Runner) initDeviceInfo() error {
-	const errPrefix = "cannot set device information: "
+func (run *Runner) findTimeLowerBound() error {
+	timeLowerBoundSources := []string{
+		// uc16
+		filepath.Join(dirs.SnapSeedDir, "seed.yaml"),
+		// uc20+
+		dirs.SnapModeenvFile,
+	}
+	// add all model files from uc20 seeds
+	allModels, err := filepath.Glob(filepath.Join(dirs.SnapSeedDir, "systems/*/model"))
+	if err != nil {
+		return err
+	}
+	timeLowerBoundSources = append(timeLowerBoundSources, allModels...)
+
+	// use all files as potential time inputs
+	for _, p := range timeLowerBoundSources {
+		info, err := os.Stat(p)
+		if os.IsNotExist(err) {
+			continue
+		}
+		if err != nil {
+			return err
+		}
+		run.moveTimeLowerBound(info.ModTime())
+	}
+	return nil
+}
 
+func findBrandAndModel() (string, string, error) {
+	if osutil.FileExists(dirs.SnapModeenvFile) {
+		return findBrandAndModel20()
+	}
+	return findBrandAndModel16()
+}
+
+func findBrandAndModel20() (brand string, model string, err error) {
+	cfg := goconfigparser.New()
+	cfg.AllowNoSectionHeader = true
+	if err := cfg.ReadFile(dirs.SnapModeenvFile); err != nil {
+		return "", "", err
+	}
+	brandAndModel, err := cfg.Get("", "model")
+	if err != nil {
+		return "", "", err
+	}
+	l := strings.SplitN(brandAndModel, "/", 2)
+	if len(l) != 2 {
+		return "", "", fmt.Errorf("cannot find brand/model in modeenv model string %q", brandAndModel)
+	}
+
+	return l[0], l[1], nil
+}
+
+func findBrandAndModel16() (string, string, error) {
```

Comment:
This could use named arguments, especially with both having the same type
```suggestion
func findBrandAndModel16() (brand, model string, err error) {
```